src/libraries/System.Private.CoreLib/src/System/IO/Path.cs

```diff
@@ -926,5 +926,56 @@ private static string GetRelativePath(string relativeTo, string path, StringComp
         /// Returns true if the path ends in a directory separator.
         /// </summary>
         public static bool EndsInDirectorySeparator(string path) => PathInternal.EndsInDirectorySeparator(path);
+
+        /// <summary>
+        /// Removes any redundant segments found in the specified path.
+        /// </summary>
+        /// <param name="path">A string containing an absolute or relative path that may or may not contain redundant segments.</param>
+        /// <returns><paramref name="path" /> without any redundant segments.</returns>
+        public static string RemoveRedundantSegments(string path)
+        {
+            if (path == null)
+                throw new ArgumentNullException(nameof(path));
+
+            if (PathInternal.IsEffectivelyEmpty(path))
+                throw new ArgumentException(SR.Arg_PathEmpty, nameof(path));
+
+            return RemoveRedundantSegments(path.AsSpan());
+        }
+
+        /// <summary>
+        /// Removes any redundant segments found in the specified path.
+        /// </summary>
+        /// <param name="path">A read-only span containing an absolute or relative path that may or may not contain redundant segments.</param>
+        /// <returns><paramref name="path" /> without any redundant segments.</returns>
+        public static string RemoveRedundantSegments(ReadOnlySpan<char> path)
+        {
+            if (PathInternal.IsEffectivelyEmpty(path))
+                throw new ArgumentException(SR.Arg_PathEmpty, nameof(path));
+
+            Span<char> destination = stackalloc char[path.Length];
+
+            PathInternal.TryRemoveRedundantSegments(path, destination, out int charsWritten);
+
+            return destination.Slice(0, charsWritten).ToString();
```

Comment:
We should avoid creating a new string if input contains no redundant segments.

Reply:
@Anipik
This is the method that receives a `ROS<char>` as parameter, and returns a `string`. If `charsWritten` is zero, I would still not have a choice but to return `path.ToString()`.

I could apply your suggestion on the other `RemoveRedundantSegments` method, the one that receives a `string` and returns a `string`: If I call `TryRemoveRedundantSegments` there, and `charsWritten` is zero, I can return the `path` that was passed as parameter, untouched.

Reply:
If the output size equals the input size then you can pass back the original string.